# Projet Introduction to BIG DATA
Configuration des librairies:
- Pandas (Extraction et analyse des données)
- Numpy (Graphiques et courbes)
- Matplotlib

In [16]:
# Configuration des librairies
%matplotlib inline
import numpy as np
import pandas as pd

# Lecture du fichier
Ce fichier csv volumineux (162,5 Mo) contient toutes les données à analyser

In [17]:
data_frame = pd.read_csv("consommation-electrique-par-secteur-dactivite-commune.csv", sep=";")

# Vision globale du fichier
Ici, on a une vision globale de toutes les colonnes avec leurs données respectives 

In [18]:
pd.read_csv("consommation-electrique-par-secteur-dactivite-commune.csv", sep=";")

Année                  Nom commune  Code commune  \
0        2011    La Chapelle-Saint-Maurice         74060   
1        2015                    Thaumiers         18261   
2        2013           Neuville-sur-Saône         69143   
3        2011    Lachapelle-sous-Rougemont         90058   
4        2016                        Seyre         31546   
5        2012                    Bourdeaux         26056   
6        2014                       Bonnée         45039   
7        2016                      Étreval         54185   
8        2015                       Aulnoy         77013   
9        2012              Causse-et-Diège         12257   
10       2012                      Marcoux         42136   
11       2016           Gensac-de-Boulogne         31218   
12       2016                     Senlisse         78590   
13       2012                       Unzent          9319   
14       2012                  Arpheuilles         18013   
15       2011            Kirsch-lès-Sierck         57364   
16       2013                      Arthezé         72009   
17       2013                     Langouet         35146   
18       2014            Neuilly-le-Vendin         53164   
19       2016                        Lapte         43114   
20       2014                  Ousse-Suzan         40215   
21       2015                       Thiers         63430   
22       2012           Graves-Saint-Amant         16297   
23       2016                     Neuville         63252   
24       2016       Lac-des-Rouges-Truites         39271   
25       2016          Beauvais-sur-Tescou         81024   
26       2012                 Coursegoules          6050   
27       2013                Graimbouville         76314   
28       2012                     Échenans         25210   
29       2015                         Ossé         35209   
...       ...                          ...           ...   
201933   2013                    Montfermy         63238   
201934   2016     Saint-André-de-Valborgne         30231   
201935   2012                       Hoedic         56085   
201936   2011                La Daguenière         49117   
201937   2011                      Molinot         21420   
201938   2013                     Godisson         61192   
201939   2015                    Fouzilhon         34105   
201940   2015                     Étainhus         76250   
201941   2012           Bourg-Sainte-Marie         52063   
201942   2014                  Longperrier         77259   
201943   2014                 Les Genettes         61187   
201944   2011  Saint-Maurice-sur-Vingeanne         21562   
201945   2016                Saint-Ferréol         74234   
201946   2015             Rigny-sur-Arroux         71370   
201947   2016         Épineuil-le-Fleuriel         18089   
201948   2011             Cauville-sur-Mer         76167   
201949   2015                    Les Haies         69097   
201950   2013                        Sauve         30311   
201951   2016                Criel-sur-Mer         76192   
201952   2013                      Busloup         41028   
201953   2014     La Villeneuve-en-Chevrie         78668   
201954   2015             Muides-sur-Loire         41155   
201955   2011                     Écharcon         91204   
201956   2012                Perdreauville         78484   
201957   2013                       Nançay         18159   
201958   2011                     Plumelec         56172   
201959   2014              Condé-sur-Risle         27167   
201960   2014                       Golbey         88209   
201961   2011         Saint-Mamert-du-Gard         30281   
201962   2015         Le Hommet-d'Arthenay         50248   

                                                 Nom EPCI  Code EPCI  \
0                    CC de la Rive Gauche du Lac d'Annecy  247400732   
1                                            CC le Dunois  241800424   
2                                              CU de Lyon  246900245   

In [24]:
# Informations détaillées du tableau de données (data frame)
data_frame.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 201963 entries, 0 to 201962
Data columns (total 43 columns):
Année                                     201963 non-null int64
Nom commune                               201963 non-null object
Code commune                              201963 non-null int64
Nom EPCI                                  201963 non-null object
Code EPCI                                 201963 non-null object
Type EPCI                                 201963 non-null object
Nom département                           201963 non-null object
Code département                          201963 non-null int64
Nom région                                201963 non-null object
Code région                               201963 non-null int64
Nb sites Résidentiel                      174131 non-null float64
Conso totale Résidentiel (MWh)            174131 non-null float64
Conso moyenne Résidentiel (MWh)           174131 non-null float64
Nb sites Professionnel                    174

# Liste des colonnes
On va extraire depuis le csv la liste des noms des colonnes du tableau

In [9]:
list(data_frame)

['Année',
 'Nom commune',
 'Code commune',
 'Nom EPCI',
 'Code EPCI',
 'Type EPCI',
 'Nom département',
 'Code département',
 'Nom région',
 'Code région',
 'Nb sites Résidentiel',
 'Conso totale Résidentiel (MWh)',
 'Conso moyenne Résidentiel (MWh)',
 'Nb sites Professionnel',
 'Conso totale Professionnel (MWh)',
 'Conso moyenne Professionnel (MWh)',
 'Nb sites Agriculture',
 'Conso totale Agriculture (MWh)',
 'Nb sites Industrie',
 'Conso totale Industrie (MWh)',
 'Nb sites Tertiaire',
 'Conso totale Tertiaire (MWh)',
 'Nb sites Secteur non affecté',
 'Conso totale Secteur non affecté (MWh)',
 "Nombre d'habitants",
 'Taux de logements collectifs',
 'Taux de résidences principales',
 'Superficie des logements < 30 m2',
 'Superficie des logements 30 à 40 m2',
 'Superficie des logements 40 à 60 m2',
 'Superficie des logements 60 à 80 m2',
 'Superficie des logements 80 à 100 m2',
 'Superficie des logements > 100 m2',
 'Résidences principales avant 1919',
 'Résidences principales de 1919 

# Le fichier à analyser
- 201963 lignes
- 43 colonnes

Avec la fonction describe() sur le tableau de données (data frame), on explore la structure du fichier et on a ainsi les résultats des fonctions suivantes sur les 43 colonnes du tableau:
- count()
- sum()
- std()
- min()
- 25%
- 50%
- 75%
- max()

In [22]:
data_frame.describe()

Année   Code commune  Code département    Code région  \
count  201963.000000  201963.000000     201963.000000  201963.000000   
mean     2013.490867   45250.310854         44.999554      52.030085   
std         1.704241   25186.596570         25.177943      24.062562   
min      2011.000000    1001.000000          1.000000      11.000000   
25%      2012.000000   24394.000000         24.000000      28.000000   
50%      2013.000000   46214.000000         46.000000      44.000000   
75%      2015.000000   64427.000000         64.000000      76.000000   
max      2016.000000   95690.000000         95.000000      93.000000   

       Nb sites Résidentiel  Conso totale Résidentiel (MWh)  \
count          1.741310e+05                    1.741310e+05   
mean           7.915993e+02                    3.989305e+03   
std            8.696828e+03                    2.848242e+04   
min            0.000000e+00                    0.000000e+00   
25%            1.290000e+02                    7.601002e+02   
50%            2.470000e+02                    1.553031e+03   
75%            5.430000e+02                    3.439311e+03   
max            1.361346e+06                    4.627239e+06   

       Conso moyenne Résidentiel (MWh)  Nb sites Professionnel  \
count                    174131.000000           174132.000000   
mean                          6.129561              113.317001   
std                           1.464097             1538.291395   
min                           0.000000                0.000000   
25%                           5.193854               20.000000   
50%                           6.152051               34.000000   
75%                           7.083365               72.000000   
max                          25.634289           244735.000000   

       Conso totale Professionnel (MWh)  Conso moyenne Professionnel (MWh)  \
count                      1.741320e+05                      174132.000000   
mean                       1.097599e+03                           9.091434   
std                        1.383726e+04                           3.334213   
min                        0.000000e+00                           0.000000   
25%                        1.546886e+02                           6.803469   
50%                        3.033256e+02                           8.923016   
75%                        7.165631e+02                          11.030172   
max                        2.298164e+06                         217.148377   

                   ...               Superficie des logements 80 à 100 m2  \
count              ...                                      201963.000000   
mean               ...                                          26.312132   
std                ...                                           6.802233   
min                ...                                           0.000000   
25%                ...                                          22.093049   
50%                ...                                          26.492603   
75%                ...                                          30.693061   
max                ...                                          80.000000   

       Superficie des logements > 100 m2  Résidences principales avant 1919  \
count                      201963.000000                      201963.000000   
mean                           48.579402                          30.909932   
std                            13.857292                          18.197286   
min                             0.000000                           0.000000   
25%                            39.583452                          16.923036   
50%                            48.648649                          28.491620   
75%                            57.851232                          42.608696   
max                           100.000000                         100.000000   

       Résidences principales de 1919 à 1945  \
count                          201963.000000  

# Analyse générale sur les EPCI
EPCI: Établissement Public de Coopération Intercommunale
9 types différents:
- CC: Communautés de communes
- CU: Communautés urbaines
- nd: Non défini
- CA: Communautés d'agglomération
- ME: Métropoles
- SAN: Syndicats d'agglomération nouvelle
- ZZ: ??
- Métropole
- SA: Syndicats d'aglomération

In [35]:
data_frame['Type EPCI'].unique()

array(['CC', 'CU', 'nd', 'CA', 'ME', 'SAN', 'ZZ', 'Métropole', 'SA'], dtype=object)

In [34]:
data_frame['Type EPCI'].value_counts()

CC           171718
CA            23716
nd             2851
CU             2116
ME             1271
SAN             104
Métropole        92
ZZ               80
SA               15
Name: Type EPCI, dtype: int64